packages/selected-network-controller/src/SelectedNetworkController.ts

```diff
@@ -147,6 +140,23 @@ export class SelectedNetworkController extends BaseControllerV2<
     domain: Domain,
     networkClientId: NetworkClientId,
   ) {
+    const networkClient = this.messagingSystem.call(
+      'NetworkController:getNetworkClientById',
+      networkClientId,
+    );
+    const networkProxy = this.#proxies.get(domain);
+    if (networkProxy === undefined) {
+      this.#proxies.set(domain, {
+        provider: createEventEmitterProxy(networkClient.provider),
+        blockTracker: createEventEmitterProxy(networkClient.blockTracker, {
+          eventFilter: 'skipInternal',
+        }),
+      });
+    } else {
+      networkProxy.provider.setTarget(networkClient.provider);
+      networkProxy.blockTracker.setTarget(networkClient.blockTracker);
+    }
+
```

Comment:
does it make sense to only update `state.domains` here and rely on `getProviderAndBlockTracker` to initialize the proxies?

Reply:
hmm do you mean instead of using `this.#proxies.get(domain);`, we use `this. getProviderAndBlockTracker(domain)`, and then we dont have to duplicate the creation of the proxies? 

i like

Reply:
oh yeah the reason to not do that - it ends up calling `NetworkController:getNetworkClientById` twice, and then in the event that its being created for the first time, it would then have 'setTarget' called on it again. I figured its best to just leave it 'undry' in this case.

Reply:
lets revist this later if thats ok!